docs/workflow/argocd.md

```diff
@@ -0,0 +1,235 @@
+repository should include claims that can be resolved by the environment classes
+we created. If you want to use an example application, you can use Wordpress in
+[this repository][wordpress-dir].
+
+### Create an ArgoCD Project for Applications
+
+Similar to the `infra` project above, create an ArgoCD project to contain
+applications that will make use of our infrastructure. All of the resources that
+we deploy in this project should be namespace-scoped, so there is no need to
+whitelist any cluster-scoped resources. An example of an `apps` project with the
+required permissions would look as follows:
+
+![alt argo-proj-apps](argo-proj-apps.png)
+
+### Configure an ArgoCD Application
+
+Now to deploy the application, create a new application in the `apps` project.
+It is important to pay special attention to the namespace because our
+application resources will be namespace-scoped. Configure your application to
+looks as follows or as needed for your application:
+
+![alt argo-apps-wordpress-1](argo-apps-wordpress-1.png)
+
+![alt argo-apps-wordpress-2](argo-apps-wordpress-2.png)
+
+After creation of the application, you should see all resources appear in the
+visualization. You can click on the `Service` resource to get the external IP
+address of the Wordpress application when it becomes available.
+
+![alt argo-wordpress-service](argo-wordpress-service.png)
+
+If you navigate to the IP address, you should see the Wordpress start page.
+
+![alt argo-wordpress-landing](argo-wordpress-landing.png)
+
+## Portable App Deployment to Multiple Clouds
+
+We have successfully deployed Wordpress on GCP with `dev` tier infrastructure.
+However, it may be desireable to provision on `prod` tier infrastructure, or
+even a different cloud provider. Assuming you also set up AWS infrastructure
+configuration in the [infrastructure](#configure-dev-and-prod-environments)
+section above, this can be accomplished with simple changes to the Wordpress
+deployment configuration. The resource claims use label selectors to be
+scheduled to a certain class. If you take a look at [the patch in our
+application configuration][app-patch], you will notice the following stanza:
+
+```yaml
+ classSelector:
+    matchLabels:
+      env: dev
+      provider: gcp
+```
+
+Simply changing these labels to `env: prod` or `provider: aws` will cause the
+resources to be provisioned on production tier infrastructure or on AWS instead
+of GCP! Furthermore, if you would like more granular options for deploying
+infrastructure, you can customize the labels on the resource classes that were
+created to include any other distinguishing traits.
+
+## Clean Up
+
+To clean up all of our application components, you can simply delete each
+application and those resources will be removed from your cluster.
+
+## Next Steps
+
+Check out one of the other continuous delivery guides or customize your
+environments for your specific use case!
```

Comment:
@prasek do you want to add a link to Gitlab example here when it is ready?

Reply:
@hasheddan yes, will do after you land this with a follow-on PR, thanks!

Reply:
actually will add it to the workflow.md